be/src/olap/field.h

```diff
@@ -450,6 +451,29 @@ uint32_t Field::hash_code(const CellType& cell, uint32_t seed) const {
     return _type_info->hash_code(cell.cell_ptr(), seed);
 }
 
+class MapField : public Field {
+public:
+    explicit MapField(const TabletColumn& column) : Field(column) {}
+
+    void consume(RowCursorCell* dst, const char* src, bool src_null, MemPool* mem_pool,
+                 ObjectPool* agg_pool) const override {
+        dst->set_is_null(src_null);
+        if (src_null) {
+            return;
+        }
+        _type_info->deep_copy(dst->mutable_cell_ptr(), src, mem_pool);
+    }
+
+        // make variable_ptr memory allocate to cell_ptr as MapValue
```

Comment:
maybe need use the code formatter to format code.

Reply:
I will try again